sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -458,6 +458,10 @@ object DataType {
   private[sql] def equalsIgnoreCompatibleCollation(from: DataType, to: DataType): Boolean = {
     (from, to) match {
       // String types with possibly different collations are compatible.
+      case (CharType(l1), CharType(l2)) => l1 == l2
+      case (VarcharType(l1), VarcharType(l2)) => l1 == l2
+      case (CharType(_) | VarcharType(_), _: StringType) => false
+      case (_: StringType, CharType(_) | VarcharType(_)) => false
       case (_: StringType, _: StringType) => true
```

Comment:
@stefankandic does this seem ok to you? I do not want to break delta. Is this only used in ddl operations?